### PR TITLE
fix: ts parsing error

### LINKED ISSUE #932: TypeScript parsing error
Minimal repro:

```js
enum SomeEnum {
    FOO = "FOO",
};
const someVariable: { [key in SomeEnum] } = { FOO: 3 };
```

Error:
```
  × Expected `:` but found `}`
   ╭─[./foo.ts:3:1]
 3 │ };
 4 │ const someVariable: { [key in SomeEnum] } = { FOO: 3 };
   ·                                         ┬
   ·                                         ╰── `:` expected
   ╰────
```

The syntax in the example uses an implicit any - so it would fail type-checking if `noImplicitAny` is enabled; but... it's still syntactically valid, and the type checking is fine if you don't have that setting enabled.

oxlint fails to parse it.

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
Fixed https://github.com/web-infra-dev/oxc/issues/932

### CODE REVIEW

File: crates/oxc_parser/src/ts/types.rs [hunk 1]
```diff
@@ -656,9 +656,7 @@ impl<'a> Parser<'a> {
             _ => TSMappedTypeModifierOperator::None,
         };
 
-        self.expect(Kind::Colon)?;
-        let type_annotation = self.parse_ts_type()?;
-
+        let type_annotation = self.parse_ts_type_annotation()?;
```
Comment (IWANABETHATGUY) [inline crates/oxc_parser/src/ts/types.rs:659]: https://github.com/microsoft/TypeScript/blob/284cd452450128439b2aaf50a5ac1d92c7fc07d1/src/compiler/parser.ts#L4398-L4399

File: crates/oxc_parser/examples/test.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+
+const someVariable: { [key in SomeEnum] } = { FOO: 3 };
```
Comment (IWANABETHATGUY) [inline crates/oxc_parser/examples/test.ts:2]: The case in #932 can be parsed now, but I am sure where should I add a new test case.

**Comment by DonIsaac** [comment]
@IWANABETHATGUY thanks for the PR! Go ahead and update conformance snapshots and I think we're good to go

File: crates/oxc_parser/examples/test.ts (hunk 2 above)
Comment (DonIsaac) [reply crates/oxc_parser/examples/test.ts:2]: I'm not certain you need it, the conformance tests should be enough

**Comment by IWANABETHATGUY** [comment]
> @IWANABETHATGUY thanks for the PR! Go ahead and update conformance snapshots and I think we're good to go

Done

**Review by Boshen — APPROVED** [review]

**Pull request merged**